core/admin/start.py

```diff
@@ -18,6 +18,36 @@
     log.info("Creating initial admin accout %s@%s with mode %s",account,domain,mode)
     os.system("flask mailu admin %s %s '%s' --mode %s" % (account, domain, password, mode))
 
+def test_DNS():
+    import dns.resolver
+    import dns.exception
+    import dns.flags
+    import dns.rdtypes
+    import dns.rdatatype
+    import dns.rdataclass
+    import time
+    # DNS stub configured to do DNSSEC enabled queries
+    resolver = dns.resolver.Resolver()
+    resolver.use_edns(0, 0, 1232)
+    resolver.flags = dns.flags.AD | dns.flags.RD
+    nameservers = resolver.nameservers
+    for ns in nameservers:
+        resolver.nameservers=[ns]
+        error = True
+        while error:
+            try:
+                result = resolver.query('example.org', dns.rdatatype.A, dns.rdataclass.IN, lifetime=10)
+                if not result.response.flags & dns.flags.AD:
+                    log.critical("Your DNS resolver at %s isn't doing DNSSEC validation; Please install unbound.", ns)
+                else:
+                    error = False
+                    continue
```

Comment:
You could use `return` here and `while True:` above  instead of the variable `error`.
Like this:
```
while True:
    try:
        result = resolver.query('example.org', dns.rdatatype.A, dns.rdataclass.IN, lifetime=10)
    except Exception as e:
        log.critical("Your DNS resolver at %s is not working (%s). Please use another resolver or enable unbound via setup.mailu.io.", ns, e);
    else:
        if result.response.flags & dns.flags.AD:
            return
        log.critical("Your DNS resolver at %s isn't doing DNSSEC validation. Please use another resolver or enable unbound via setup.mailu.io.", ns)
    time.sleep(5)
```

Reply:
That would change the logic though, wouldn't it?

Here I was trying to ensure that **all** listed nameservers are DNSSEC capable... since we don't control which one will be queried

Reply:
Yes sure - so it would be `break` instead of `return`.

Reply:
the `error` variable looks indeed unnecesary … we want to loop over all resolver, regardless of what happened in one of them before… if you have 4 nameservers configured an 2 are not DNSSEC-capable, just log them once they are reached in the loop and continue with the next one
Right?

Reply:
so more like:

```python
while True:
  try:
    result = resolver.query('example.org', dns.rdatatype.A, dns.rdataclass.IN, lifetime=10)
  except Exception as e:
    log.critical("Your DNS resolver at %s is not working (%s). Please use another resolver or enable unbound via setup.mailu.io.", ns, e)
  else:
    if not result.response.flags & dns.flags.AD:
      log.critical("Your DNS resolver at %s isn't doing DNSSEC validation. Please use another resolver or enable unbound via setup.mailu.io.", ns)
  time.sleep(5)
```

Reply:
Nope. This would be infinite.
_all_ Nameservers need to be functioning and doing DNSSEC - so a break (ie "continue testing next server") when a server is "good" is the correct thing to do.

Reply:
That's a2f6243